src/devices/KeyMatrix/KeyMatrix.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.KeyMatrix
+{
+    /// <summary>
+    /// GPIO key matrix Driver
+    /// </summary>
+    public class KeyMatrix : IDisposable
+    {
+        /// <summary>
+        /// Get output pins
+        /// </summary>
+        public IEnumerable<int> OutputPins => _outputPins;
+
+        /// <summary>
+        /// Get input pins
+        /// </summary>
+        public IEnumerable<int> InputPins => _inputPins;
+
+        /// <summary>
+        /// Get all buttons' values
+        /// </summary>
+        public ReadOnlySpan<PinValue> Values => _buttonValues.AsSpan();
+
+        /// <summary>
+        /// Get or set interval in milliseconds
+        /// </summary>
+        public int ScanInterval { get; set; }
+
+        /// <summary>
+        /// Get buttons' values by output
+        /// </summary>
+        /// <param name="output">Output index</param>
+        public ReadOnlySpan<PinValue> this[int output] => _buttonValues.AsSpan(output * _outputPins.Length, _inputPins.Length);
+
+        private int[] _outputPins;
+        private int[] _inputPins;
+        private GpioController _gpioController;
+        private PinValue[] _buttonValues;
+        private bool _pinsOpened;
+        private int _currentOutput = 0;
+
+        /// <summary>
+        /// Initialize key matrix
+        /// </summary>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="outputPins">Output pins</param>
+        /// <param name="inputPins">Input pins</param>
+        /// <param name="scanInterval">Scanning interval in milliseconds</param>
+        public KeyMatrix(GpioController gpioController, IEnumerable<int> outputPins, IEnumerable<int> inputPins, int scanInterval)
```

Comment:
Wondering if you want as well to implement a constructor with creating automatically a GpioController? See here https://github.com/dotnet/iot/tree/master/src/devices#gpiocontroller